collect_app/src/main/java/org/odk/collect/android/utilities/Regex.java

```diff
@@ -0,0 +1,43 @@
+package org.odk.collect.android.utilities;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Created by simran on 4/8/2017.
+ */
+
+public class Regex {
+    public static final String GPS_LOCATION= "^-?[0-9]+\\.[0-9]+\\s-?[0-9]+\\.[0-9]+\\s-?[0-9]+\\"
+            + ".[0-9]+\\s[0-9]+\\.[0-9]+$";
+    public static final String VALID_GOOGLE_SHEETS_ID = "^[a-zA-Z0-9\\-]+$";
+    private static Pattern valid_google_sheets_id, gps_location;
+
+    /**
+     * Google sheets currently only allows a-zA-Z0-9 and dash
+     */
+    // Check to see if answer is a location, if so, get rid of accuracy
+    // and altitude
+    // try to match a fairly specific pattern to determine
+    // if it's a location
+    // [-]#.# [-]#.# #.# #.#
+
+
+    static {
+        valid_google_sheets_id = Pattern.compile(VALID_GOOGLE_SHEETS_ID);
```

Comment:
And I actually think that as we discussed in the issue, both the expressions and the methods that will use them should be in the classes they are used in rather than in a separate regex class. What do you think @dcbriccetti @nribeka?